src/lib/storage/dictionary_segment/attribute_vector_iterable.hpp

```diff
@@ -51,8 +59,10 @@ class AttributeVectorIterable : public PointAccessibleSegmentIterable<AttributeV
   class Iterator : public BaseSegmentIterator<Iterator<ZsIteratorType>, SegmentPosition<ValueID>> {
    public:
     using ValueType = ValueID;
-    explicit Iterator(const ValueID null_value_id, ZsIteratorType attribute_it, ChunkOffset chunk_offset)
-        : _null_value_id{null_value_id}, _attribute_it{attribute_it}, _chunk_offset{chunk_offset} {}
+    explicit Iterator(const ValueID null_value_id, ZsIteratorType attribute_it, ChunkOffset chunk_offset,
```

Comment:
Gilt für alle Iteratoren:
Wenn das Event "on_iterator_dereference" nicht aufgezeichnet werden soll/muss, muss das Segment nicht dem Iterator übergeben werden.

Reply:
Lass da mal in Ruhe drüber reden. Aktuell ist glaube ich SortedSegmentSearch der einzige Operator, der den Iterator nicht vollständig nutzt.

Reply:
on_iterator_dereference is way too expensive. We don't need it anymore.